pandas/io/parsers.py

```diff
@@ -867,71 +867,23 @@ def __init__(self, f, engine=None, **kwds):
         else:
             engine = "python"
             engine_specified = False
-
+        self.engine = engine
         self._engine_specified = kwds.get("engine_specified", engine_specified)
 
         if kwds.get("dialect") is not None:
             dialect = kwds["dialect"]
             if dialect in csv.list_dialects():
                 dialect = csv.get_dialect(dialect)
+            kwds = self._refresh_kwargs_based_on_dialect(kwds, dialect)
```

Comment:
merge master as some of this updated; and rename this like check_defaults_dialect (or similar)